ios/FluentUI/People Picker/ContactView.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+@objc(MSFContactView)
+open class ContactView: UIView {
+    private struct Constants {
+        static let contactWidth: CGFloat = 70.0
+        static let avatarViewWidth: CGFloat = 70.0
+        static let avatarViewHeight: CGFloat = 70.0
+        static let labelMinimumHeight: CGFloat = 16.0
+        static let titleLabelMaximumHeight: CGFloat = 28.0
+        static let subtitleMaximumHeight: CGFloat = 24.0
+        static let spacingBetweenAvatarAndLabelContainer: CGFloat = 13.0
+    }
+
+    @objc public var avatarImage: UIImage? {
+        didSet {
+            if let subtitleLabel = subtitleLabel {
+                setupAvatarView(with: titleLabel.text ?? "", and: subtitleLabel.text ?? "")
+            } else {
+                setupAvatarView(with: titleLabel.text ?? "")
+            }
+        }
+    }
+
+    private let avatarView: AvatarView
+    private var titleLabel: UILabel
+    private var subtitleLabel: UILabel?
+    private var labelContainer: UIView
+
+    /// Initializes the contact view by creating an avatar view with a first name and last name
+    ///
+    /// - Parameters:
+    ///   - title: String that will be the text of the top label
+    ///   - subtitle: String that will be the text of the bottom label
+    @objc public convenience init(title: String, subtitle: String) {
+        self.init(title: title, subtitle: subtitle, identifier: nil)
+    }
+
+    /// Initializes the contact view by creating an avatar view with an identifier
+    ///
+    /// - Parameters:
+    ///   - identifier: String that will be used to identify the contact (e.g. email, phone number, first name)
+    @objc public convenience init(identifier: String) {
+        self.init(title: nil, subtitle: nil, identifier: identifier)
+    }
+
+    private init(title: String?, subtitle: String?, identifier: String?) {
+        avatarView = AvatarView(avatarSize: .extraExtraLarge, withBorder: false, style: .circle)
+        labelContainer = UIView(frame: .zero)
+        titleLabel = UILabel(frame: .zero)
+        super.init(frame: .zero)
+
+        if let title = title, let subtitle = subtitle {
+            setupAvatarView(with: title, and: subtitle)
+            setupSubtitleLabel(using: subtitle)
+            setupTitleLabel(using: title)
+        } else if let identifier = identifier {
+            setupAvatarView(with: identifier)
+            setupTitleLabel(using: identifier)
+        }
+
+        backgroundColor = Colors.surfacePrimary
+        setupLayout()
+    }
+
+    public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    private func setupAvatarView(with firstName: String, and lastName: String) {
+        let fullName = firstName + " " + lastName
```

Comment:
>fullName = firstName + " " + lastName [](start = 12, length = 37)

let's call this primaryText, and change the params to title and subtitle. We can't assume that clients passed a name as title and a last name as a subtitle.

I'm actually not sure if it would be correct to assume that we can concatenate both in this way

Reply:
Does the avatar view really need a primaryText set? If not, maybe we should avoid setting it since we're not sure

---
In reply to: [448534319](https://github.com/microsoft/fluentui-apple/pull/101#discussion_r448534319) [](ancestors = 448534319)

Reply:
> let's call this primaryText, and change the params to title and subtitle. We can't assume that clients passed a name as title and a last name as a subtitle.
> 
> I'm actually not sure if it would be correct to assume that we can concatenate both in this way

I see what you mean regarding the concatenation. Do you have any suggestions on how to go about passing a first and last name as the `primaryText` argument to `avatarView.setup()`?

Reply:
The `AvatarView` doesn't actually require a `primaryText` as it is an `Optional`. What do you mean by "avoid setting it since we're not sure if the text is correct"? I think we need to set pass in a `primaryText` argument when we call `avatarView.setup()` in order to display the one/two text labels.

Reply:
This actually makes me think of an issue we might run into for the next Phase of development (the collection of Contact Cards). Clients will need to know the distinction between a name and a last name (hopefully People APIs know that distinction, it would be terrible if clients need to do text splits haha). To handle a person object, we use the Persona class in Fluent, which only has a fullName property. I imagine we'll use the same object to hold the data source for contact cards, we might need to modify that to accommodate for a first name and last name in that class.

Reply:
> > Does the avatar view really need a primaryText set? If not, maybe we should avoid setting it since we're not sure if the text is correct
> > In reply to: [448534319](https://github.com/microsoft/fluentui-apple/pull/101#discussion_r448534319) [](ancestors = 448534319)
> 
> The `AvatarView` doesn't actually require a `primaryText` as it is an `Optional`. What do you mean by "avoid setting it since we're not sure if the text is correct"? I think we need to set pass in a `primaryText` argument when we call `avatarView.setup()` in order to display the one/two text labels.

Oh! Of course, avatarView needs that to setup the initials image, when there's no actual image... I just took a look at the method that creates the initials "initialLetters" and they just take the first letter from the first and second word on the fullname, regardless of whether they are actually the name and last name of a person. Fascinating, not the cleanest in my opinion.

So, I guess it is ok that we pass that concatenation as primaryText to avatarView